### PR TITLE
test: improve coverage for transaction_controller.go

### PR DESCRIPTION
**Pull request by antoinegelloz**

### CODE REVIEW

File: pkg/api/controllers/transaction_controller.go [hunk 1]
```diff
@@ -192,40 +193,49 @@ func (ctl *TransactionController) RevertTransaction(c *gin.Context) {
...
 	}
 
 	txId, err := strconv.ParseUint(c.Param("txid"), 10, 64)
 	if err != nil {
-		c.AbortWithStatus(http.StatusBadRequest)
+		ResponseError(c, ledger.NewValidationError("invalid transaction ID"))
 		return
 	}
 
-	if err := l.(*ledger.Ledger).SaveMeta(c.Request.Context(), core.MetaTargetTypeTransaction, txId, m); err != nil {
```
Comment (gfyrag) [inline pkg/api/controllers/transaction_controller.go:216]: By design, adding metadata on a non existent transaction could exists before that.
Maybe it is ok but we have to check.

File: pkg/ledger/ledger.go [hunk 2]
```diff
@@ -99,7 +99,15 @@ func (l *Ledger) CountTransactions(ctx context.Context, m ...storage.TxQueryModi
 }
 
 func (l *Ledger) GetTransaction(ctx context.Context, id uint64) (core.Transaction, error) {
-	return l.store.GetTransaction(ctx, id)
+	tx, err := l.store.GetTransaction(ctx, id)
+	if err != nil {
+		return core.Transaction{}, err
+	}
+	if len(tx.Postings) == 0 {
```
Comment (gfyrag) [inline pkg/ledger/ledger.go]: I think we should modify the storage implementation to return an error if the transaction is not found.

File: pkg/ledger/ledger.go [hunk 3]
```diff
@@ -115,10 +123,16 @@ func (l *Ledger) LoadMapping(ctx context.Context) (*core.Mapping, error) {
...
+func (l *Ledger) RevertTransaction(ctx context.Context, id uint64) (core.Transaction, error) {
 	tx, err := l.store.GetTransaction(ctx, id)
 	if err != nil {
-		return nil, err
+		return core.Transaction{}, err
+	}
+	if len(tx.Postings) == 0 {
+		return core.Transaction{}, NewNotFoundError("transaction not found")
+	}
+	if tx.IsReverted() {
```
Comment (gfyrag) [inline pkg/ledger/ledger.go:134]: Cool!

File: pkg/api/controllers/transaction_controller.go [hunk 4]
```diff
@@ -192,40 +193,49 @@ func (ctl *TransactionController) RevertTransaction(c *gin.Context) {
...
+	_, err = l.(*ledger.Ledger).GetTransaction(c.Request.Context(), txId)
```
Comment (gfyrag) [inline pkg/api/controllers/transaction_controller.go:218]: It is a breaking change. We need to validate this.

**Review by gfyrag — CHANGES_REQUESTED** [review]
Comments embedded

File: pkg/api/controllers/transaction_controller.go (hunk 4 above)
Comment (jdupas22) [reply pkg/api/controllers/transaction_controller.go:218]: what are we doing here exactly ? i'm not sure i understand

Comment (gfyrag) [reply pkg/api/controllers/transaction_controller.go:218]: Before this changes, we can add a metadata on a transaction while it is not existings.
It is ok on accounts, but not on a transaction. It is more an unexpected behavior than a breaking changes finally.

File: pkg/ledger/ledger.go (hunk 2 above)
Comment (antoinegelloz) [reply pkg/ledger/ledger.go]: I changed the signature of the store method, it now returns `*core.Transaction`, with a `nil` response if the transaction isn't found. Is it better ?

File: pkg/api/controllers/transaction_controller.go (hunk 4 above)
Comment (antoinegelloz) [reply pkg/api/controllers/transaction_controller.go:218]: I agree :) It is clear now @jdupas22?

**Review by flemzord — APPROVED** [review]

**Review by gfyrag — APPROVED** [review]

**Review by gfyrag — APPROVED** [review]

**Pull request merged**